server/odc-service/src/main/java/com/oceanbase/odc/config/HookConfiguration.java

```diff
@@ -91,4 +108,31 @@ private void approvalFlowConfigUsageCheck(Long flowConfigId) {
         }
     }
 
+    private void projectReferenceCheck(Long userId, Long organizationId) {
+        Map<Long, Set<ResourceRoleName>> projectId2ResourceRoleNames =
+                resourceRoleService.getProjectId2ResourceRoleNames(userId).entrySet().stream()
+                        .filter(e -> e.getValue().contains(ResourceRoleName.OWNER)
+                                || e.getValue().contains(ResourceRoleName.DBA))
+                        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+        if (projectId2ResourceRoleNames.size() == 0) {
+            return;
+        }
+        Map<Long, ProjectEntity> id2Project =
+                projectRepository.findAllByOrganizationId(organizationId).stream().filter(p -> p.getArchived() == false)
+                        .collect(Collectors.toMap(ProjectEntity::getId, p -> p));
+        String names = projectId2ResourceRoleNames.keySet().stream().map(id -> id2Project.get(id).getName())
+                .collect(Collectors.joining(", "));
+
+        String errorMessage = String.format(
+                "User id=%s cannot be deleted because it has been referenced to following project: {%s}", userId,
+                names);
+
+        throw new UnsupportedException(ErrorCodes.CannotOperateDueReference,
```

Comment:
BadRequestException will be better？
UnsupportedException means that a feature is not supported yet

Reply:
done